### PR TITLE
Feature #2328 - create PDS solution for phan 

### PR DESCRIPTION
**Pull request by zigfridus**
Initial commit for #2328

### CODE REVIEW

**Comment by zigfridus** [comment]
@Jeeppler 
Please have a look at this PR

File: sechub-pds-solutions/phan/README.adoc [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+= Phan + PDS
+
+Phan is a static analyzer for PHP that prefers to minimize false-positives. Phan attempts to prove incorrectness rather than correctness.
+In addition, the combination of Phan and PDS allows both to run in a Kubernetes cluster or on virtual machines.
+
+This folder contains the scripts necessary to run Phan Wrapper + PDS locally within a container. It also contains scripts to create the PDS + Phan container and push it to your container registry, as well as a helm chart to install and run Phan Wrapper + PDS in a Kubernetes cluster.
+
+== Current limitations
+
+- only docker images uploaded as `.tar` files can be scanned
```
Comment (Jeeppler) [inline sechub-pds-solutions/phan/README.adoc:15]: Please remove this.

File: sechub-pds-solutions/phan/README.adoc
Comment (Jeeppler) [inline sechub-pds-solutions/phan/README.adoc:1]: Please remove all the parts which do not apply.

File: sechub-pds-solutions/phan/README.adoc [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+==== Scan script
+
+The folder contains a start script which does the manual steps for you:
+
+----
+./01-start-single-docker-compose.sh
+----
+
+==== Together with SecHub
```
Comment (Jeeppler) [inline sechub-pds-solutions/phan/README.adoc:43]: Please remove this section as it is not accurate.

File: sechub-pds-solutions/phan/composer.json
Comment (Jeeppler) [inline sechub-pds-solutions/phan/composer.json:1]: Please move this file into `phan/docker`.

File: sechub-pds-solutions/phan/composer.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+    "require": {
+        "phan/phan": "^5.4"
```
Comment (Jeeppler) [inline sechub-pds-solutions/phan/composer.json:3]: Use the exact version number. 

Furthermore, it would be nice to have this version number injected via variable (docker compose). You can achieve this by using a placeholder like "PHAN_VERSION" and replacing it with `sed` and the value from the variable.

```suggestion
        "phan/phan": "5.4"
```

File: sechub-pds-solutions/phan/composer.lock
Comment (Jeeppler) [inline sechub-pds-solutions/phan/composer.lock:1]: Is this file necessary? 

The composer.json file should be enough as composer will download `phan` and all it's dependencies inside the container.

File: sechub-pds-solutions/phan/docker/Phan-Debian.dockerfile [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+# The image argument needs to be placed on top
+ARG BASE_IMAGE
+FROM ${BASE_IMAGE}
+
+# The remaining arguments need to be placed after the `FROM`
+# See: https://ryandaniels.ca/blog/docker-dockerfile-arg-from-arg-trouble/
+
+LABEL org.opencontainers.image.source="https://github.com/mercedes-benz/sechub"
+LABEL org.opencontainers.image.title="SecHub Phan+PDS Image"
+LABEL org.opencontainers.image.description="A container which combines Phan Wrapper with the SecHub Product Delegation Server (PDS)"
```
Comment (Jeeppler) [inline sechub-pds-solutions/phan/docker/Phan-Debian.dockerfile:12]:
```suggestion
LABEL org.opencontainers.image.description="A container which combines Phan with the SecHub Product Delegation Server (PDS)"
```

File: sechub-pds-solutions/phan/docker/Phan-Debian.dockerfile [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+LABEL maintainer="SecHub FOSS Team"
+
+USER root
+
+# Build Args
+ARG PHAN_WRAPPER_VERSION="5.4.3"
```
Comment (Jeeppler) [inline sechub-pds-solutions/phan/docker/Phan-Debian.dockerfile:18]: A wrapper is only necessary in case of cloud services/proprietary tools which cannot run locally. Phan can run inside the container just fine.

```suggestion
ARG PHAN_VERSION="5.4.3"
```

File: sechub-pds-solutions/phan/README.adoc (hunk 1 above)
Comment (zigfridus) [reply sechub-pds-solutions/phan/README.adoc:15]: Done.

File: sechub-pds-solutions/phan/README.adoc
Comment (zigfridus) [reply sechub-pds-solutions/phan/README.adoc:1]: Done.

File: sechub-pds-solutions/phan/README.adoc (hunk 2 above)
Comment (zigfridus) [reply sechub-pds-solutions/phan/README.adoc:43]: Done.

File: sechub-pds-solutions/phan/composer.json
Comment (zigfridus) [reply sechub-pds-solutions/phan/composer.json:1]: Done.

**Comment by Jeeppler** [comment]
@zigfridus Please request a re-review once you are done with most changes.

File: sechub-pds-solutions/phan/composer.lock
Comment (zigfridus) [reply sechub-pds-solutions/phan/composer.lock:1]: Removed it

File: sechub-pds-solutions/phan/docker/Phan-Debian.dockerfile (hunk 5 above)
Comment (zigfridus) [reply sechub-pds-solutions/phan/docker/Phan-Debian.dockerfile:18]: Done.

File: sechub-pds-solutions/phan/docker/Phan-Debian.dockerfile (hunk 4 above)
Comment (zigfridus) [reply sechub-pds-solutions/phan/docker/Phan-Debian.dockerfile:12]: Done

File: sechub-pds-solutions/phan/composer.json (hunk 3 above)
Comment (zigfridus) [reply sechub-pds-solutions/phan/composer.json:3]: Please have a look at my next commit about this improvement. I'm not sure if I resolved this problem correctly.

**Review by Jeeppler — APPROVED** [review]
Very well done 🥇

- Code is well done
- I tested the solution it works as expected
